src/io/image_iter_common.h

```diff
@@ -337,21 +337,11 @@ struct ImageDetNormalizeParam :  public dmlc::Parameter<ImageDetNormalizeParam>
 struct PrefetcherParam : public dmlc::Parameter<PrefetcherParam> {
   /*! \brief number of prefetched batches */
   size_t prefetch_buffer;
-  /*! \brief data type */
-  dmlc::optional<int> dtype;
 
   // declare parameters
   DMLC_DECLARE_PARAMETER(PrefetcherParam) {
     DMLC_DECLARE_FIELD(prefetch_buffer).set_default(4)
         .describe("Maximum number of batches to prefetch.");
-    DMLC_DECLARE_FIELD(dtype)
```

Comment:
dtype is problematic for PrefetcherIter but I'm afraid this change will break API. If someone use an data iter with dtype = float32, it will throw an exception.